airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java

```diff
@@ -641,6 +642,11 @@ void testIncrementalSync() throws Exception {
     final JobInfoRead jobInfoRead = apiClient.getConnectionApi().resetConnection(new ConnectionIdRequestBody().connectionId(connectionId));
     waitWhileJobHasStatus(apiClient.getJobsApi(), jobInfoRead.getJob(),
         Sets.newHashSet(JobStatus.PENDING, JobStatus.RUNNING, JobStatus.INCOMPLETE, JobStatus.FAILED));
+    // This is a band-aid to prevent some race conditions where the job status was updated but we may
+    // still be cleaning up some data in the reset table. This would be an argument for reworking the
+    // source of truth of the replication workflow state to be in DB rather than in Memory and
+    // serialized automagically by temporal
+    waitWhileJobIsRunning(apiClient.getJobsApi(), jobInfoRead.getJob(), Duration.ofMinutes(1));
```

Comment:
@gosusnp - do we need this line at all? now that you've updated the handler won't `waitWhileJobHasStatus` (2 lines above) also get the correct source of truth? it seems redundant.

Reply:
I believe those tests were flaky until this PR got merged.
Those two checks are different, the first one is looking at the DB state, this one is looking at temporal state. Until we fix the actual endpoint (either change how we handle clean up after a reset or have a single source of truth for state), yes, I think we still need this check.

Reply:
ah. i see. they are using different endpoints. can you make sure we have an issue to follow up on this, please? it's pretty spooky.

Reply:
Captured in https://github.com/airbytehq/airbyte/issues/19854